nibabel/streamlines/tractogram.py

```diff
@@ -0,0 +1,543 @@
+        """ Internal dictionary that makes sure data are 2D array. """
+
+        def __setitem__(self, key, value):
+            value = np.asarray(value)
+
+            if value.ndim == 1 and value.dtype != object:
+                # Reshape without copy
+                value.shape = ((len(value), 1))
+
+            if value.ndim != 2:
+                raise ValueError("data_per_streamline must be a 2D array.")
+
+            # We make sure there is the right amount of values
+            # (i.e. same as the number of streamlines in the tractogram).
+            if len(value) != len(self.tractogram):
+                msg = ("The number of values ({0}) should match the number of"
+                       " streamlines ({1}).")
+                raise ValueError(msg.format(len(value), len(self.tractogram)))
+
+            self.store[key] = value
+
+    class DataPerPointDict(DataDict):
+        """ Internal dictionary making sure data are :class:`ArraySequence` objects. """
+
+        def __setitem__(self, key, value):
+            value = ArraySequence(value)
+
+            # We make sure we have the right amount of values (i.e. same as
+            # the total number of points of all streamlines in the tractogram).
+            if len(value._data) != len(self.tractogram.streamlines._data):
+                msg = ("The number of values ({0}) should match the total"
+                       " number of points of all streamlines ({1}).")
+                nb_streamlines_points = self.tractogram.streamlines._data
+                raise ValueError(msg.format(len(value._data),
+                                            len(nb_streamlines_points)))
+
+            self.store[key] = value
+
+    def __init__(self, streamlines=None,
+                 data_per_streamline=None,
+                 data_per_point=None):
+        """
+        Parameters
+        ----------
+        streamlines : list of ndarray of shape (Nt, 3) (optional)
+            Sequence of T streamlines. One streamline is an ndarray of
+            shape (Nt, 3) where Nt is the number of points of streamline t.
+        data_per_streamline : dict of list of ndarray of shape (P,) (optional)
+            Sequence of T ndarrays of shape (P,) where T is the number of
+            streamlines defined by `streamlines`, P is the number of
+            properties associated to each streamline.
+        data_per_point : dict of list of ndarray of shape (Nt, M) (optional)
+            Sequence of T ndarrays of shape (Nt, M) where T is the number
+            of streamlines defined by `streamlines`, Nt is the number of
+            points for a particular streamline t and M is the number of
+            scalars associated to each point (excluding the three
+            coordinates).
+        """
+        self.streamlines = streamlines
+        self.data_per_streamline = data_per_streamline
+        self.data_per_point = data_per_point
+        self._affine_to_rasmm = np.eye(4)
```

Comment:
No no, don't apology, the code changed but comments/doctstrings might not have! See my previous comment for an explanation.